data_key.go

```
@@ -5,30 +5,46 @@ import (
-// Values passed to Key must not be modified after insertion as the tree may
-// retain and deduplicate them.
-//
-// The returned byte slice is not stored. You may use the same backing
-// array between calls.
-type KeyGenerator interface {
-	Key(mmdbtype.DataType) ([]byte, error)
+// dataHasher serializes a value into its canonical, pointer-free MMDB encoding
+// and hashes it for the in-memory data map. Hash matches are always followed by
+// an exact comparison, so correctness does not depend on collision resistance.
+type dataHasher struct {
+	bytes.Buffer
```

golangci-lint (2.12.2)

[error] there must be an empty line separating embedded fields from regular fields

(embeddedstructfieldcheck)